### PR TITLE
Dala 3610 lksg update subcontracting companies

### PR DESCRIPTION
**Pull request by stewal19**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server
- [x] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithLksg.cy.ts [hunk 1]
```diff
@@ -92,14 +92,33 @@ describe("Component test for the LksgPanel", () => {
...
-      cy.get("th");
       cy.get("th").eq(0).should("have.text", "Procurement Category");
       cy.get("th").eq(1).should("have.text", "Procured Products/Services");
       cy.get("th").eq(2).should("have.text", "Number of Direct Suppliers and Countries");
       cy.get("th").eq(3).should("have.text", "Order Volume");
     });
   });
 
+  it("Validate that the subcategories countries modal is displayed correctlz and contains the correct headers", () => {
+    const preparedFixture = getPreparedFixture("lksg-with-subcontracting-countries", preparedFixtures);
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithLksg.cy.ts:103]: corrected

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgRiskPositionMultiSelectComponent.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.dataland.frameworktoolbox.frameworks.lksg.custom
+
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.MultiSelectComponent
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+
+/**
+ * Represents the LksgRiskPositionMultiSelectComponent component
+ */
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgRiskPositionMultiSelectComponent.kt:10]: changed

File: dataland-frontend/src/components/forms/parts/fields/LksgSubcontractingCompaniesFormField.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
+<template>
+  <!--  todo remove border-nones -->
+  <div class="form-field border-none">
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-frontend/src/components/forms/parts/fields/LksgSubcontractingCompaniesFormField.vue]: Sollte laut sonar weg sein

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgRiskPositionMultiSelectComponentFactory.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * Generates NaceCodesComponents from rows with the component "LkSG Risk Position"
+ */
+@Component
+class LksgRiskPositionMultiSelectComponentFactory(@Autowired val templateDiagnostic: TemplateDiagnostic) :
+    TemplateComponentFactory {
+    override fun canGenerateComponent(row: TemplateRow): Boolean =
+        row.component == "LkSG Risk Position MultiSelect Component"
+
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgRiskPositionMultiSelectComponentFactory.kt:21]: canGenerateComponent verwendet überall das "=". 
Ansonsten lass uns drei gerne darüber telefonieren

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgSubcontractingCompaniesComponent.kt [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        )
+    }
+
+    override fun generateDefaultFixtureGenerator(sectionBuilder: FixtureSectionBuilder) {
+        requireDocumentSupportIn(setOf(NoDocumentSupport))
+        val fixtureExperssion = if (isNullable) {
+            "dataGenerator.valueOrNull(dataGenerator.generateSubcontractingCompanies())"
+        } else {
+            "dataGenerator.generateSubcontractingCompanies()"
+        }
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgSubcontractingCompaniesComponent.kt:91]: Nicht ganz sicher was hier gemeint ist. Wenn isNullable = false, dann möchte ich ja sichergehen, dass ich auch einen nicht Null return habe.

File: dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifApiAccessor.kt [hunk 6]
```diff
@@ -51,7 +52,7 @@ class GleifApiAccessor(
     fun getFullIsinMappingFile(targetFile: File) {
         logger.info("Successfully acquired download link for mapping")
         val tempZipFile = File.createTempFile("gleif_mapping_update", ".zip")
-        downloadIndirectFile(URL(isinMappingReferenceUrl), tempZipFile)
+        downloadIndirectFile(URI(isinMappingReferenceUrl).toURL(), tempZipFile)
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifApiAccessor.kt:55]: Weil URL deprecated ist, ist in Sonar aufgetaucht, daher wurde es geändert

File: dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifApiAccessor.kt [hunk 7]
```diff
@@ -88,7 +89,7 @@ class GleifApiAccessor(
 
     private fun downloadFileFromGleif(urlSuffx: String, targetFile: File, fileDescription: String) {
         logger.info("Starting download of $fileDescription.")
-        val downloadUrl = URL("$gleifBaseUrl/$urlSuffx")
+        val downloadUrl = URI("$gleifBaseUrl/$urlSuffx").toURL()
         downloadFile(downloadUrl, targetFile)
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifApiAccessor.kt:93]: Gleicher Grund wie an der anderen URL Stelle

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgSubcontractingCompaniesComponentFactory.kt [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * Generates the SubContractingCompanies component
+ */
+@Component
+class LksgSubcontractingCompaniesComponentFactory(@Autowired val templateDiagnostic: TemplateDiagnostic) :
+    TemplateComponentFactory {
+    override fun canGenerateComponent(row: TemplateRow): Boolean =
+        row.component == "LkSG Subcontracting Companies"
+
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgSubcontractingCompaniesComponentFactory.kt:21]: Siehe den anderen Kommentar zum Funktionsbody

File: dataland-frontend/src/components/forms/parts/fields/LksgSubcontractingCompaniesFormField.vue [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      :options="allCountries"
+      optionLabel="label"
+      v-model:selectedItemsBindInternal="selectedCountries"
+      inputClass="long"
+    />
+  </div>
+  <div class="form-field">
+    <FormKit type="group" :name="name" v-model="industriesPerCountry">
+      <div v-for="(el, index) in selectedCountries" :key="el.label">
+        <div v-if="index > 0" class="h-2rem" />
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-frontend/src/components/forms/parts/fields/LksgSubcontractingCompaniesFormField.vue:14]: changed

File: dataland-frontend/src/components/forms/parts/fields/LksgSubcontractingCompaniesFormField.vue [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    },
+  },
+  methods: {
+    /**
+     * Remove item from selected countries list
+     * @param countryCode - Country code to be removed
+     */
+    removeItemFromListOfSelectedCountries(countryCode: string) {
+      this.selectedCountries = this.selectedCountries.filter((el) => countryCode !== el.value);
+    },
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-frontend/src/components/forms/parts/fields/LksgSubcontractingCompaniesFormField.vue:82]: changed

File: dataland-frontend/src/components/resources/dataTable/conversion/lksg/LksgDisplayValueGetters.ts [hunk 11]
```diff
@@ -160,6 +188,44 @@ export function formatLksgProcurementCategoriesForDisplay(
...
+  input: { [key: string]: Array<string> } | null | undefined,
+  fieldLabel: string,
+): AvailableMLDTDisplayObjectTypes {
+  let convertedValueForModal = null;
+  if (!input) {
+    return MLDTDisplayObjectForEmptyString;
+  } else {
+    convertedValueForModal = convertLksgSubcontractingCompaniesToListForModal(input);
+  }
+
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-frontend/src/components/resources/dataTable/conversion/lksg/LksgDisplayValueGetters.ts:209]: Schau ich mir gleich an

[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-frontend/src/components/resources/dataTable/conversion/lksg/LksgDisplayValueGetters.ts:208]: Ist geändert

File: dataland-frontend/tests/component/components/forms/CreateLksgDataset.cy.ts [hunk 12]
```diff
@@ -0,0 +1,101 @@
+import CreateLksgDataset from "@/components/forms/CreateLksgDataset.vue";
+import { minimalKeycloakMock } from "@ct/testUtils/Keycloak";
+import { type FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+import { type CompanyAssociatedDataLksgData, type LksgData } from "@clients/backend";
+import { submitButton } from "@sharedUtils/components/SubmitButton";
+
+describe("test YesNoBaseDataPointFormField for entries", () => {
+  let preparedFixtures: Array<FixtureData<LksgData>>;
```
Comment (jhaerDfine) [inline dataland-frontend/tests/component/components/forms/CreateLksgDataset.cy.ts:8]: renaming "test ..." (zumindest zu Test ...)

**Pull request merged**